lib/src/ruby/ruby_value.cc

```diff
@@ -272,4 +272,10 @@ namespace facter { namespace ruby {
         emitter << Null;
     }
 
+    ruby_value const* ruby_value::wrap_child(VALUE child) const {
+        auto result = new ruby_value(child);
+        children.emplace_back(result);
```

Comment:
Should be done in the emplace_back call; if someone mistakenly adds an early return before it is emplaced, it will leak.

```
children.emplace_back(new ruby_value(child));
return children.back();
```